http-filter-example/http_filter_integration_test.cc

```diff
@@ -10,17 +14,35 @@ class HttpFilterSampleIntegrationTest : public HttpIntegrationTest,
    * Initializer for an individual integration test.
    */
   void SetUp() override {
-    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
-    registerPort("upstream_0", fake_upstreams_.back()->localAddress()->ip()->port());
-    createTestServer("http-filter-example/envoy.conf", {"http"});
+    HttpIntegrationTest::SetUp();
+    initialize();
   }
 
-  /**
-   * Destructor for an individual integration test.
-   */
-  void TearDown() override {
-    test_server_.reset();
-    fake_upstreams_.clear();
+  void createUpstreams() override {
+    HttpIntegrationTest::createUpstreams();
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+    ports_.push_back(fake_upstreams_.back()->localAddress()->ip()->port());
+  }
+
+  void initialize() override {
+    config_helper_.addFilter(
+        "{ name: sample, config: {} }");
+
+    config_helper_.addConfigModifier([](envoy::api::v2::Bootstrap &bootstrap) {
+      auto *sample_cluster = bootstrap.mutable_static_resources()->add_clusters();
```

Comment:
I think it probably would have been better to point at test/integration/integration_test.h` rather than the HTTP2 test, sorry about that. You can remove this `addConfigModifier` if you also remove the `createUpstreams`, which is only needed for H2 backends. We only care about simple H1 in this example.